libraries/Filter/HarmonicNotchFilter.h

```diff
@@ -0,0 +1,61 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#pragma once
+
+#include <AP_Math/AP_Math.h>
+#include <cmath>
+#include <inttypes.h>
+#include <AP_Param/AP_Param.h>
+#include "NotchFilter.h"
+
+template <class T>
+class HarmonicNotchFilter {
+public:
+    // set parameters
+    void create(uint8_t harmonics);
+    void init(float sample_freq_hz, float center_freq_hz, float bandwidth_hz, float attenuation_dB);
+    T apply(const T &sample);
+    void reset();
+    void update(float center_freq_hz);
+    ~HarmonicNotchFilter();
+
+private:
+    NotchFilter<T>*  _filters;
+    float _sample_freq_hz;
+    float _A;
```

Comment:
It would be very helpful for future developers to have comments beside each member declaration giving some idea of what these are for.

Reply:
done